addons/account/models/account_account.py

```diff
@@ -737,20 +737,21 @@ def _order_accounts_by_frequency_for_partner(self, company_id, partner_id, move_
         return self._get_most_frequent_accounts_for_partner(company_id, partner_id, move_type)
 
     @api.model
-    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
-        if not name and self._context.get('partner_id') and self._context.get('move_type'):
-            return self._order_accounts_by_frequency_for_partner(
-                            self.env.company.id, self._context.get('partner_id'), self._context.get('move_type'))
-        domain = domain or []
-        if name:
-            if operator in ('=', '!='):
-                name_domain = ['|', ('code', '=', name.split(' ')[0]), ('name', operator, name)]
-            else:
-                name_domain = ['|', ('code', '=like', name.split(' ')[0] + '%'), ('name', operator, name)]
-            if operator in expression.NEGATIVE_TERM_OPERATORS:
-                name_domain = ['&', '!'] + name_domain[1:]
-            domain = expression.AND([name_domain, domain])
-        return self._search(domain, limit=limit, order=order)
+    def _search_display_name(self, operator, value):
+        if not value and self._context.get('partner_id') and self._context.get('move_type'):
+            ids = self._order_accounts_by_frequency_for_partner(
+                self.env.company.id, self._context.get('partner_id'), self._context.get('move_type'))
+            return [('id', 'in', ids)]
+        if not value:
+            return []
```

Comment:
So searching with `[('display_name', '=', False)]` is equivalent to searching with `[]`?
That doesn't make sense to me.